source/includes/fundamentals/code-examples/Indexes.cs

```diff
@@ -0,0 +1,206 @@
+    {
+        Console.WriteLine("single index");
+
+        // begin-single-index
+        collection.Indexes.CreateOne("{ title : 1 }");
+        // end-single-index
+
+        // begin-single-index-query
+        // Define query parameters
+        var filter = Builders<Movie>.Filter.Eq("title", "Batman");
+        var sort = Builders<Movie>.Sort.Ascending("title");
+        var projection = Builders<Movie>.Projection.Include("title").Exclude("_id");
+
+        // Execute query
+        var results = collection.Find(filter).Sort(sort).Project(projection);
+        // end-single-index-query
+    }
+
+    private static void CompoundIndex(IMongoCollection<Movie> collection)
+    {
+        Console.WriteLine("compound index");
+
+        // begin-compound-index
+        collection.Indexes.CreateOne("{ title : 1, rated: 1 }");
+        // end-compound-index
+
+        // begin-compound-index-query
+        // Define query parameters
+        var typeFilter = Builders<Movie>.Filter.Eq("type", "movie");
+        var ratedFilter = Builders<Movie>.Filter.Eq("rated", "G");
+        var filter = Builders<Movie>.Filter.And(typeFilter, ratedFilter);
+        var sort = Builders<Movie>.Sort.Ascending("{ title : 1, rated: 1 }");
+        var projection = Builders<Movie>.Projection.Include("type").Include("rated").Exclude("_id");
+
+        // Execute query
+        var results = collection.Find(filter).Sort(sort).Project(projection);
+        // end-compound-index-query
+    }
+
+    private static void MultiKeyIndex(IMongoCollection<Movie> collection)
+    {
+        Console.WriteLine("multi-key index");
+
+        // begin-multi-key-index
+        collection.Indexes.CreateOne("{ rated: 1, genres: 1, title: 1 }");
+        // end-multi-key-index
+
+        // begin-multi-key-query
+        // Define query parameters
+        var genreFilter = Builders<Movie>.Filter.Eq("genres", "Animation");
+        var ratedFilter = Builders<Movie>.Filter.Eq("rated", "G");
+        var filter = Builders<Movie>.Filter.And(genreFilter, ratedFilter);
+        var sort = Builders<Movie>.Sort.Ascending("title");
+        var projection = Builders<Movie>.Projection.Include("title").Include("rated").Exclude("_id");
+
+        // Execute query
+        var results = collection.Find(filter).Sort(sort).Project(projection);
+        // end-multi-key-query
+    }
+
+    private static void TextIndex(IMongoCollection<Movie> collection)
+    {
+        Console.WriteLine("text index");
+
+        // begin-text-index
+        try
+        {
+            collection.Indexes.CreateOne("{ plot : \"text\" }");
+        }
+        // Prints a message if a text index already exists with a different configuration 
+        catch (MongoCommandException e)
+        {
+            if (e.CodeName == "IndexOptionsConflict")
+            {
+                Console.WriteLine("There is an existing text index with different options");
+            }
+        }
+        // end-text-index
```

Comment:
Is there a reason you're including the try/catch block on this example but not on the others? If it's not explicitly necessary I think we should just show the line that creates the index so that it mirrors the other examples.

Reply:
Can update this one and the geospatial examples to only include the index creation line.